### ISSUE TITLE
External Monitor -- "declaration failed. \\\"OK\\\" unknown property"

### ISSUE THREAD

**Issue by devildog-tech** [F5Networks/f5-appsvcs-extension#596]
### Environment
Application Services Version: 3.35.0
BIG-IP Version: 16.1.2.1:

### Summary
When creating an external monitor with a environmental variable of "200 OK" fails with the following error message.

`"msg": "declaration failed. \\\"OK\\\" unknown property",` 

Posting the following JSON will result in the above error.

```
{
    "$schema": "https://raw.githubusercontent.com/F5Networks/f5-appsvcs-extension/master/schema/latest/as3-schema.json",
    "class": "AS3",
    "action": "deploy",
    "persist": true,
    "id": "example.com",
    "declaration": {
        "class": "ADC",
        "schemaVersion": "3.0.0",
        "example.com": {
            "class": "Tenant",
            "Shared": {
                "class": "Application",
                "template": "shared",
                "enable": true,
                "example.com_external_monitor": {
                    "monitorType": "external",
                    "interval": 5,
                    "timeUntilUp": 0,
                    "timeout": 16,
                    "script": {
                        "base64": "IyEvYmluL3NoDQpJUD1gZWNobyAkezF9IHwgc2VkICdzLzo6ZmZmZjovLydgDQpQT1JUPSR7Mn0NCg0KUElERklMRT0iL3Zhci9ydW4vYGJhc2VuYW1lICR7MH1gLiR7SVB9XyR7UE9SVH0ucGlkIg0KIyBraWxsIG9mIHRoZSBsYXN0IGluc3RhbmNlIG9mIHRoaXMgbW9uaXRvciBpZiBodW5nIGFuZCBsb2cgY3VycmVudCBwaWQNCmlmIFsgLWYgJFBJREZJTEUgXQ0KdGhlbg0KICAgZWNobyAiRUFWIGV4Y2VlZGVkIHJ1bnRpbWUgbmVlZGVkIHRvIGtpbGwgJHtJUH06JHtQT1JUfSIgfCBsb2dnZXIgLXAgbG9jYWwwLmVycm9yDQogICBraWxsIC05IGBjYXQgJFBJREZJTEVgID4gL2Rldi9udWxsIDI+JjENCmZpDQplY2hvICIkJCIgPiAkUElERklMRQ0KDQpjdXJsLWFwZCAtayAtaSAtLWNvbm5lY3QtdGltZW91dCA0IC0tcmVzb2x2ZSAke0ZRRE59OiR7UE9SVH06JHtJUH0gaHR0cHM6Ly8ke0ZRRE59JHtVUkl9IHxncmVwIC1pICIke1JFQ1Z9IiA+IC9kZXYvbnVsbCAyPiYxDQoNCmlmIFsgJD8gLWVxIDAgXQ0KdGhlbg0KICAgIHJtIC1mICRQSURGSUxFDQogICAgZWNobyAiVVAiDQplbHNlDQogICAgcm0gLWYgJFBJREZJTEUNCmZpDQpleGl0"
                    },
                    "environmentVariables": {
                        "RECV": "200 OK",
                        "URI": "/example.aspx",
                        "FQDN": "example.com"
                    },
                    "remark": "EXAMPLE",
                    "class": "Monitor"
                }
            }
        }
    }
}
```

If you post the same JSON but remove the `"OK"` from `"RECV": "200 OK",` you will receive a successful, example below. 

```
{
    "$schema": "https://raw.githubusercontent.com/F5Networks/f5-appsvcs-extension/master/schema/latest/as3-schema.json",
    "class": "AS3",
    "action": "deploy",
    "persist": true,
    "id": "example.com",
    "declaration": {
        "class": "ADC",
        "schemaVersion": "3.0.0",
        "example.com": {
            "class": "Tenant",
            "Shared": {
                "class": "Application",
                "template": "shared",
                "enable": true,
                "example.com_external_monitor": {
                    "monitorType": "external",
                    "interval": 5,
                    "timeUntilUp": 0,
                    "timeout": 16,
                    "script": {
                        "base64": "IyEvYmluL3NoDQpJUD1gZWNobyAkezF9IHwgc2VkICdzLzo6ZmZmZjovLydgDQpQT1JUPSR7Mn0NCg0KUElERklMRT0iL3Zhci9ydW4vYGJhc2VuYW1lICR7MH1gLiR7SVB9XyR7UE9SVH0ucGlkIg0KIyBraWxsIG9mIHRoZSBsYXN0IGluc3RhbmNlIG9mIHRoaXMgbW9uaXRvciBpZiBodW5nIGFuZCBsb2cgY3VycmVudCBwaWQNCmlmIFsgLWYgJFBJREZJTEUgXQ0KdGhlbg0KICAgZWNobyAiRUFWIGV4Y2VlZGVkIHJ1bnRpbWUgbmVlZGVkIHRvIGtpbGwgJHtJUH06JHtQT1JUfSIgfCBsb2dnZXIgLXAgbG9jYWwwLmVycm9yDQogICBraWxsIC05IGBjYXQgJFBJREZJTEVgID4gL2Rldi9udWxsIDI+JjENCmZpDQplY2hvICIkJCIgPiAkUElERklMRQ0KDQpjdXJsLWFwZCAtayAtaSAtLWNvbm5lY3QtdGltZW91dCA0IC0tcmVzb2x2ZSAke0ZRRE59OiR7UE9SVH06JHtJUH0gaHR0cHM6Ly8ke0ZRRE59JHtVUkl9IHxncmVwIC1pICIke1JFQ1Z9IiA+IC9kZXYvbnVsbCAyPiYxDQoNCmlmIFsgJD8gLWVxIDAgXQ0KdGhlbg0KICAgIHJtIC1mICRQSURGSUxFDQogICAgZWNobyAiVVAiDQplbHNlDQogICAgcm0gLWYgJFBJREZJTEUNCmZpDQpleGl0"
                    },
                    "environmentVariables": {
                        "RECV": "200",
                        "URI": "/example.aspx",
                        "FQDN": "example.com"
                    },
                    "remark": "EXAMPLE",
                    "class": "Monitor"
                }
            }
        }
    }
}
```

I suspect it's really not as much about the `OK` as it is how the F5 is interpreting spaces in the `RECV:`. But I have not had a chance to test it out. Unless I'm doing something wrong.

We use several of these types of monitors and is preventing us from deploying them with AS3.

**Comment by dstokesf5** [comment]
Thank you for your feedback. I have confirmed the bug and added it to our internal product backlog as AUTOTOOL-3139.

As you suspected, it appears to be related to the spaces.

**Comment by MicSkr** [comment]
Heads up. Ran into the same issue today. Same circumstances, the external monitor variable value has a space for "200 OK".

Also tried, but same OK message and no value in monitor configured:
"200%20OK" (URL Encoded Space)
"200/u0020OK" (UTF-16 Encoded Space)

**Comment by dstokesf5** [comment]
This has been resolved in the 3.38.0 release of AS3.
